### PR TITLE
[FEATURE] Search history

### LINKED ISSUE #94: [BQ] What are the most searched for items (words) in the search bar?

### PR DESCRIPTION
**Pull request by juanyepesp**
CLOSES #94 

- This saves in local storage using a .json file the search history of the user, while showing it to them. 
- It only does so when the user hits the "search" button on the keyboard

https://github.com/ISIS3510-202410-Team23/SwiftApp/assets/77981728/16772789-a7e8-428a-8e99-be26182b3a72

### CODE REVIEW

**Review by mar-cas3 — DISMISSED** [review]
Great, thank you!!
beyonce kick GIF

**Comment by Laurarestrepo03** [comment]
I have a question. I'm on my phone so I've only skimmed the code, but I'm curious to know when a search is added to history, when clicking search or when clicking on a spot after searching?

**Comment by mar-cas3** [comment]
@juanyepesp just realized something, not sure if you implemented, i'll check the code and delete if so, but if you log in with another user are the recent searches reset?

**Comment by juanyepesp** [comment]
> I have a question. I'm on my phone so I've only skimmed the code, but I'm curious to know when a search is added to history, when clicking search or when clicking on a spot after searching?

@Laurarestrepo03 as I said it only saves when hitting the search, this is the way flutter has it implemented as well

**Comment by juanyepesp** [comment]
> @juanyepesp just realized something, not sure if you implemented, i'll check the code and delete if so, but if you log in with another user are the recent searches reset?
> 
> Edit: I tested locally to check and the recents are the same with different accounts. I think you should reset the file once a user logs out so this doesn't happen (seems easier than keeping file x user and makes sense that they'll be reset at this point)

@mar-cas3 should the file be reset every new log in or should the search history be saved by user?

**Comment by Laurarestrepo03** [comment]
> > I have a question. I'm on my phone so I've only skimmed the code, but I'm curious to know when a search is added to history, when clicking search or when clicking on a spot after searching?
> 
> @Laurarestrepo03 as I said it only saves when hitting the search, this is the way flutter has it implemented as well

Thanks! Sorry btw, was checking the PR on my phone and must've missed that part 😬

**Comment by juanyepesp** [comment]
> > > I have a question. I'm on my phone so I've only skimmed the code, but I'm curious to know when a search is added to history, when clicking search or when clicking on a spot after searching?
> > 
> > 
> > @Laurarestrepo03 as I said it only saves when hitting the search, this is the way flutter has it implemented as well
> 
> Thanks! Sorry btw, was checking the PR on my phone and must've missed that part 😬

No worries!! I was debating if it should be in both cases, so I might change it if I figure out how to do it!

**Comment by Laurarestrepo03** [comment]
> > @juanyepesp just realized something, not sure if you implemented, i'll check the code and delete if so, but if you log in with another user are the recent searches reset?
> > Edit: I tested locally to check and the recents are the same with different accounts. I think you should reset the file once a user logs out so this doesn't happen (seems easier than keeping file x user and makes sense that they'll be reset at this point)
> 
> @mar-cas3 should the file be reset every new log in or should the search history be saved by user?

Just squeezing in here but maybe you can do what do and reset the file with every _log out_. I think saving by user might be storing sentive information (at least that's the argument for the reason I did my features that way)

**Review by Laurarestrepo03 — CHANGES_REQUESTED** [review]
I'm requesting the same change as @mar-cas3

**Comment by juanyepesp** [comment]
> > > @juanyepesp just realized something, not sure if you implemented, i'll check the code and delete if so, but if you log in with another user are the recent searches reset?
> > > Edit: I tested locally to check and the recents are the same with different accounts. I think you should reset the file once a user logs out so this doesn't happen (seems easier than keeping file x user and makes sense that they'll be reset at this point)
> > 
> > 
> > @mar-cas3 should the file be reset every new log in or should the search history be saved by user?
> 
> Just squeezing in here but maybe you can do what I do and reset the file with every _log out_. I think saving by user might be storing sentive information (at least that's the argument for the reason I did my features that way)

@Laurarestrepo03 Alrighty, this feedback is very much appreciated! I'll look into making this changes ASAP.

**Comment by Laurarestrepo03** [comment]
> > > > @juanyepesp just realized something, not sure if you implemented, i'll check the code and delete if so, but if you log in with another user are the recent searches reset?
> > > > Edit: I tested locally to check and the recents are the same with different accounts. I think you should reset the file once a user logs out so this doesn't happen (seems easier than keeping file x user and makes sense that they'll be reset at this point)
> > > 
> > > 
> > > @mar-cas3 should the file be reset every new log in or should the search history be saved by user?
> > 
> > 
> > Just squeezing in here but maybe you can do what I do and reset the file with every _log out_. I think saving by user might be storing sentive information (at least that's the argument for the reason I did my features that way)
> 
> @Laurarestrepo03 Alrighty, this feedback is very much appreciated! I'll look into making this changes ASAP.

I edited my comment just now, be sure to check out the new version!

**Comment by juanyepesp** [comment]
@Laurarestrepo03 
>Unless you don't store by user locally and fetch it with every log in

Could you clarify what you mean by this?

**Comment by Laurarestrepo03** [comment]
> @Laurarestrepo03
> 
> > Unless you don't store by user locally and fetch it with every log in
> 
> Could you clarify what you mean by this?

Ofc, wrote it in a bit of a rush but I was thinking maybe you can save the history locally, but once the user logs out send it to firestore, that way when they log back in they can access it again. This is very **extra**  but it's just an idea if you actually wanna save by user, if not you can just delet when they log out and never recover the previous info.

**Comment by juanyepesp** [comment]
@mar-cas3 @Laurarestrepo03 With these changes, signing out will delete the document and therefore the search history.

**Review by mar-cas3 — APPROVED** [review]
Everything 🆗 !

File: FoodBookApp/UI/Views/User/UserView.swift [hunk 1]
```diff
@@ -96,6 +100,11 @@ struct UserView: View {
                         print("Failed to sign out...")
                         // TODO: show user message
                     }
+                    do {
+                        try FileManager.default.removeItem(at: fileURL)
```
Comment (Laurarestrepo03) [inline FoodBookApp/UI/Views/User/UserView.swift:104]: Nice!

**Review by Laurarestrepo03 — APPROVED** [review]

**Pull request merged**